### PR TITLE
Database and DAO added

### PR DESCRIPTION
**Pull request by ShafinMahmudCS**
Can you please review the Database and DAO before it gets merged into the development branch?

### CODE REVIEW

File: backend/src/main/java/com/MerchStore/backend/Dao/CartDao.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+            return preparedStatement.executeUpdate() == 1;
+        }catch (SQLException e){
+            System.out.println(e.getMessage());
+        }
+        return false;
+    }
+
+    @Override
+    public boolean update(Cart cart) {
+        String statement = "UPDATE cart set (cart_id, user_id) = (?, ?) where cart_id = ?";
```
Comment (Abudy72) [inline backend/src/main/java/com/MerchStore/backend/Dao/CartDao.java]: I dont think there would be a scenario where we reassign an existing cart to someone else.

File: backend/src/main/java/com/MerchStore/backend/Dao/CartListDao.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+            return preparedStatement.executeUpdate() == 1;
+        }catch (SQLException e){
+            System.out.println(e.getMessage());
+        }
+        return false;
+    }
+
+    @Override
+    public boolean update(CartList cart_list) {
+        String statement = "UPDATE cart_list set (cart_id, product_id, product_quantity) = (?, ?, ?) where cart_id = ?";
```
Comment (Abudy72) [inline backend/src/main/java/com/MerchStore/backend/Dao/CartListDao.java]: Same thing here, we wont be updating existing cart_ids, they are used to identify each row uniquely.

File: backend/src/main/java/com/MerchStore/backend/Model/Order.java [hunk 3]
```diff
@@ -5,15 +5,15 @@
...
+    private long quantity;
 
     private double totalAmount;
 
     private String orderStatus;
 
     private final long cartId;
 
-    public Order(long orderId, int quantity, double totalAmount, String orderStatus, long cartId){
+    public Order(long orderId, long quantity, double totalAmount, String orderStatus, long cartId){
```
Comment (Abudy72) [inline backend/src/main/java/com/MerchStore/backend/Model/Order.java]: I dont think someone would order a large amount of any given product, int was just fine!

File: backend/src/main/java/com/MerchStore/backend/Model/Product.java [hunk 4]
```diff
@@ -10,10 +10,10 @@ public class Product {
 
     private double price;
 
-    private int stock;
+    private long stock;
```
Comment (Abudy72) [inline backend/src/main/java/com/MerchStore/backend/Model/Product.java]: long is an overkill for stock too, i hardly doubt we would have large inventory.

**Review by Abudy72 — CHANGES_REQUESTED** [review]

File: backend/src/main/java/com/MerchStore/backend/Dao/CartDao.java (hunk 1 above)
Comment (ShafinMahmudCS) [reply backend/src/main/java/com/MerchStore/backend/Dao/CartDao.java]: I have changed the update method to throw UnsupportedOperationException.

File: backend/src/main/java/com/MerchStore/backend/Dao/CartListDao.java (hunk 2 above)
Comment (ShafinMahmudCS) [reply backend/src/main/java/com/MerchStore/backend/Dao/CartListDao.java]: I have changed the update method to just update the product quantity.

File: backend/src/main/java/com/MerchStore/backend/Model/Order.java (hunk 3 above)
Comment (ShafinMahmudCS) [reply backend/src/main/java/com/MerchStore/backend/Model/Order.java]: I have changed the data type of quantity to int.

File: backend/src/main/java/com/MerchStore/backend/Model/Product.java (hunk 4 above)
Comment (ShafinMahmudCS) [reply backend/src/main/java/com/MerchStore/backend/Model/Product.java]: I have changed the data type of stock to int.

**Review by ghost — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Review by jason-obj — APPROVED** [review]
Nothing catches my eye on the first glance, things look good great work.

LGTM

**Review by Abudy72 — APPROVED** [review]

**Pull request merged**